edweiss-app/app/(app)/lectures/slides/index.tsx

```diff
@@ -0,0 +1,208 @@
+ * @author Gustavo Maia & Adamm Alaoui
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import RouteHeader from '@/components/core/header/RouteHeader';
+import Icon from '@/components/core/Icon';
+import TActivityIndicator from '@/components/core/TActivityIndicator';
+import TText from '@/components/core/TText';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import { CollectionOf, getDownloadURL } from '@/config/firebase';
+import t from '@/config/i18config';
+import { ApplicationRoute } from '@/constants/Component';
+import { usePrefetchedDynamicDoc } from '@/hooks/firebase/firestore';
+import useListenToMessages from '@/hooks/useListenToMessages';
+import LectureDisplay from '@/model/lectures/lectureDoc';
+import { useLocalSearchParams } from 'expo-router';
+import * as ScreenOrientation from 'expo-screen-orientation';
+import React, { useEffect, useState } from 'react';
+import { ActivityIndicator, Dimensions, Linking } from 'react-native';
+import Pdf from 'react-native-pdf';
+
+// Types
+type Lecture = LectureDisplay.Lecture;
+
+
+// ------------------------------------------------------------
+// -------------------  Remote Control Screen  ----------------
+// ------------------------------------------------------------
+
+const LectureScreen: ApplicationRoute = () => {
+    const { courseNameString, lectureIdString } = useLocalSearchParams();
+    const courseName = courseNameString as string;
+    const lectureId = lectureIdString as string;
+
+    // FCM token receiver for changing pages 
+    useListenToMessages((msg) => {
+        if (msg.type == "go_to_slide") {
+            setPage(msg.slideIndex);
+        }
+    });
+
+
+    // Hooks
+    const [numPages, setNumPages] = useState<number>(0);        // Total number of pages in the PDF
+    const [page, setPage] = useState<number>(0);                // Current page, starting from 1
+    const [uri, setUri] = useState<string>('');                 // Url state
+    //const [isLandscape, setIsLandscape] = useState(false);
+    const [isFullscreen, setIsFullscreen] = useState(false);    // FullScreen display of pdf toggle
+
+    const [lectureDoc] = usePrefetchedDynamicDoc(CollectionOf<Lecture>(`courses/${courseName}/lectures`), lectureId, undefined);
+
+    // Load images when the component mounts
+    useEffect(() => {
+        if (lectureDoc) {
+            getUri();
+            setLandscape();
+        }
+
+        /* todo for a next sprint, detection of rotation change from landscape to portrait and different UI for each mode
```

Comment:
You can delete this useless comments

Reply:
Yes I will eventually remove it, it is just there for now so that I can reuse these features for a next sprint